src/strings.c

```diff
@@ -1977,6 +1977,8 @@ f_trim(typval_T *argvars, typval_T *rettv)
     if (argvars[1].v_type == VAR_STRING)
     {
 	mask = tv_get_string_buf_chk(&argvars[1], buf2);
+	if (STRLEN(mask) == 0)
```

Comment:
Instead of using STRLEN() function, you can also use `*mask == NUL` here.

Reply:
Applied.

Reply:
Seems like both GCC and clang generate identical code for both `strlen(str) == 0` and `*str == NUL` cases:

https://stackoverflow.com/a/67125598/1989046

I do not write C regularly, but to me, it seems that the `strlen(str) == 0` is a bit easier to read.
Do you still think `*mask == NUL` is preferred?

Reply:
`*mask == NUL` appears more commonly in the codebase.